FiftyOne.Pipeline.Core/FlowElements/PipelineBuilder.cs

```diff
@@ -118,25 +121,30 @@ public virtual IPipeline BuildFromConfiguration(PipelineOptions options)
 
             try
             {
-                foreach (var elementOptions in options.Elements)
-                {
+                // Create element builders in parallel
+                var flowElementQueue = new ConcurrentQueue<IFlowElement>(); 
+                Parallel.ForEach(options.Elements, (elementOptions =>
+                {
                     if (elementOptions.SubElements != null &&
                         elementOptions.SubElements.Count > 0)
                     {
                         // The configuration has sub elements so create
                         // a ParallelElements instance.
-                        AddParallelElementsToList(FlowElements, elementOptions, counter);
+                        AddParallelElementsToList(flowElementQueue, elementOptions, counter);
                     }
                     else
                     {
                         // The configuration has no sub elements so create
                         // a flow element.
-                        AddElementToList(FlowElements, elementOptions,
+                        AddElementToList(flowElementQueue, elementOptions,
                             $"element {counter}");
                     }
-                    counter++;
-                }
+                    Interlocked.Increment(ref counter);
+                }));
 
+                // Add created builders to flow elements
+                FlowElements.AddRange(flowElementQueue);
```

Comment:
We constructed the elements in parallel and they were added in the chronological order of construction completion into the  `flowElementQueue`, but we expect that they should be added into the `FlowElements` in the order in which they were specified in the configuration, not in the chronological order of construction completion...  So perhaps we need to associate an index with each of them during construction - to place into the right position..

Reply:
Used dictionary to preserve order